R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#' 
+#' @param treatment The name of the colData variable used as treatment in the model.
+#'
+#' @param mediator The name of the colData variable used as mediator in the model.
+#'   (default: \code{mediator = NULL})
+#'
+#' @param assay.type The name of the assay used for feature-wise mediation analysis.
+#'   (default: \code{assay.type = NULL})
+#' 
+#' @param dim.type The name of the reducedDim used for component-wise mediation analysis.
+#'   (default: \code{dim.type = NULL})
+#'
+#' @param family A specification for the outcome model link function.
+#'    (default: \code{family = gaussian("identity")})
+#' 
+#' @param covariates Name or list of names of colData variables used as covariates
+#'   in the model. (default: \code{covariates = NULL})
+#' 
+#' @param p.adj.method A single character value for selecting adjustment method
+#'   of p-values. Passed to `p.adjust` function. (default: \code{p.adj.method = "BH"})
+#' 
+#' @param add.metadata TRUE or FALSE, should the model metadata be returned.
+#'   (default: \code{add.metadata = FALSE})
+#' 
+#' @param verbose TRUE or FALSE, should execution messages be printed.
+#'   (default: \code{verbose = TRUE})
+#' 
+#' @param ... additional parameters that can be passed to \code{\link[mediation:mediate]{mediate}}.
+#' 
+#' @details
+#' This wrapper of \code{\link[mediation:mediate]{mediate}} for
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' provides a simple method to analyse the effect of a treatment variable on an
+#' outcome variable found in \code{colData(se)} through the mediation of either
+#' another variable in colData (argument \code{mediator}) or an assay
+#' (argument \code{assay.type}) or a reducedDim (argument \code{dim.type}). Notably,
+#' those three arguments are mutually exclusive.
+#' 
+#' @return 
+#' A \code{data.frame} of p-values, effect sizes and confidence intervals for
+#' the ACMEs and ADEs of every mediator included in the analysis. Columns include:
+#' 
+#' \describe{
+#'   \item{Treatment}{the treatment variable}
+#'   \item{Mediator}{the mediator variable}
+#'   \item{Outcome}{the outcome variable}
+#'   \item{ACME_estimate}{the Average Causal Mediation Effect (ACME) estimate}
+#'   \item{ADE_estimate}{the Average Direct Effect (ADE) estimate}
+#'   \item{ACME_pval}{the p-value for the ACME estimate}
+#'   \item{ADE_pval}{the p-value for the ADE estimate}
+#'   \item{ACME_adjpval}{the adjusted p-value for the ACME estimate}
+#'   \item{ADE_adjpval}{the adjusted p-value for the ADE estimate}
+#'   \item{ACME_CI_lower}{the lower limit of the confidence interval for the ACME estimate}
+#'   \item{ACME_CI_upper}{the upper limit of the confidence interval for the ACME estimate}
+#'   \item{ADE_CI_lower}{the lower limit of the confidence interval for the ADE estimate}
+#'   \item{ADE_CI_upper}{the upper limit of the confidence interval for the ADE estimate}
+#' }
```

Comment:
Just thinking if all of these are needed. We could simplify this and only provide the ones that are commonly used. Then the users can have all the rest in metadata, for the more dedicated users?

Reply:
Also "ADE_pval" and "ACME_pval" are unnecessary imo, users should only look at adjpval